src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -123,13 +123,7 @@ public RenameRewriter(RenameRewriterParameters parameters) : base(visitIntoStruc
                 _isProcessingComplexifiedSpans = false;
                 _skipRenameForComplexification = 0;
 
-                // TODO: These contexts are not changed for a document. ConflictResolver.Session should be refactored to cache them in a dictionary,
-                _renamedSymbolContexts = CreateSymbolKeyToRenamedSymbolContextMap(parameters.RenameSymbolContexts, SymbolKey.GetComparer());
-                _textSpanToLocationContextMap = CreateTextSpanToLocationContextMap(parameters.TokenTextSpanRenameContexts);
-                _stringAndCommentRenameContexts = GroupStringAndCommentsTextSpanRenameContexts(parameters.StringAndCommentsTextSpanRenameContexts);
-                _replacementTexts = _renamedSymbolContexts.Select(pair => pair.Value.ReplacementText).ToImmutableHashSet();
-                _originalTexts = _renamedSymbolContexts.Select(pair => pair.Value.OriginalText).ToImmutableHashSet();
-                _allPossibleConflictNames = _renamedSymbolContexts.SelectMany(pair => pair.Value.PossibleNameConflicts).ToImmutableHashSet();
+                (_textSpanToLocationContextMap, _renamedSymbolContexts, _stringAndCommentRenameContexts, _replacementTexts, _originalTexts, _allPossibleConflictNames) = parameters.DocumentRenameInfo;
```

Comment:
this is a quite heavy handed expansion :) 

Should we just hold onto `_documentRenameInfo` as the field?

Reply:
It's the one of my favorite parts of the record type : )
But I agree they can be fields